sources/packages/api/src/msfaa-integration/msfaa-request.service.ts

```diff
@@ -0,0 +1,151 @@
+      (accumulator, pendingMSFAARequest) =>
+        accumulator + parseInt(pendingMSFAARequest.student.sin),
+      0,
+    );
+
+    //Create records and create the unique file sequence number
+    let uploadResult: MSFAAUploadResult;
+    await this.sequenceService.consumeNextSequence(
+      `MSFAA_${offeringIntensity}_SENT_FILE`,
+      async (nextSequenceNumber: number, entityManager: EntityManager) => {
+        try {
+          this.logger.log("Creating MSFAA request content...");
+          // Create the Request content for the MSFAA file by populating the
+          // header, footer and trailer content.
+          const fileContent = this.msfaaService.createMSFAARequestContent(
+            msfaaRecords,
+            nextSequenceNumber,
+            totalSINHash,
+          );
+          // Create the request filename with the file path for the MSFAA Request
+          // sent File.
+          const fileInfo = await this.msfaaService.createRequestFileName(
+            offeringIntensity,
+          );
+          this.logger.log("Uploading content...");
+          uploadResult = await this.msfaaService.uploadContent(
+            fileContent,
+            fileInfo.filePath,
+          );
+
+          // Creates the repository based on the entity manager that
+          // holds the transaction already created to manage the
+          // sequence number.
+          const msfaaNumberRepo = entityManager.getRepository(MSFAANumber);
+          this.msfaaNumberService.updateRecordsInSentFile(
+            msfaaRecordIds,
+            getUTCNow(),
+            msfaaNumberRepo,
+          );
+        } catch (error) {
+          this.logger.error(
+            `Error while uploading content for ${offeringIntensity} MSFAA Request: ${error}`,
+          );
+          throw error;
+        }
+      },
+    );
+    return uploadResult;
+  }
+
+  /**
+   * Use the information on the MSFAA, referenced application
+   * student, user and institutionlocation objects are used
+   * to generate the record to be send to MSFAA request.
+   * @param pendingMSFAARecords referenced application
+   * student, user and institutionlocation information.
+   * @param offeringIntensity offeringintensity of the record.
+   * @returns MSFAA record for the student.
+   */
+  private createMSFAARecord(
+    pendingMSFAARecords: MSFAANumber,
+    offeringIntensity: string,
+  ): MSFAARecord {
+    return {
+      id: pendingMSFAARecords.id,
+      msfaaNumber: pendingMSFAARecords.msfaaNumber,
+      sin: pendingMSFAARecords.student.sin,
+      institutionCode:
+        pendingMSFAARecords.referenceApplication.offering.institutionLocation
+          .institutionCode,
+      birthDate: pendingMSFAARecords.student.birthdate,
+      surname: pendingMSFAARecords.student.user.lastName,
+      givenName: pendingMSFAARecords.student.user.firstName,
+      gender: pendingMSFAARecords.student.gender,
+      maritalStatus: "married",
```

Comment:
why `maritalStatus` is static

Reply:
https://github.com/bcgov/SIMS/pull/827#discussion_r740505848 - For now we are having it static, this will be changed when we decide on where we are fetching the marital status in the future stories. I believe I have mentioned it in my comments.

Reply:
Please add a TODO comment to mark it and make SonarCloud highlight it.